libwasmvm/src/error/go.rs

```diff
@@ -15,77 +15,185 @@ use crate::memory::UnmanagedVector;
 //
 #[repr(i32)] // This makes it so the enum looks like a simple i32 to Go
 #[derive(PartialEq)]
-pub enum GoResult {
-    Ok = 0,
+pub enum GoError {
+    None = 0,
     /// Go panicked for an unexpected reason.
     Panic = 1,
     /// Go received a bad argument from Rust
     BadArgument = 2,
-    /// Ran out of gas while using the SDK (e.g. storage)
+    /// Ran out of gas while using the SDK (e.g. storage). This can come from the Cosmos SDK gas meter
+    /// (https://github.com/cosmos/cosmos-sdk/blob/v0.45.4/store/types/gas.go#L29-L32).
     OutOfGas = 3,
-    /// An error happened during normal operation of a Go callback, which should abort the contract
-    Other = 4,
+    /// Error while trying to serialize data in Go code (typically json.Marshal)
```

Comment:
I'm not sure about re-numbering. But I guess this only matters if we have mismatched wasmvm and libwasm.so versions.

So, I guess best to do now, before cutting 1.0